From bfd3be952234ef7c16aad4bafde496052a9185de Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Tue, 23 Sep 2014 14:56:11 -0400
Subject: [drm] bochs: changes from 3.14..3.15 for bochs

Message-id: <1411484172-28849-9-git-send-email-kraxel@redhat.com>
Patchwork-id: 96488
O-Subject: [RHEL-7.1 kernel PATCH 8/9] drm changes from 3.14..3.15 for bochs
Bugzilla: 1043379
RH-Acked-by: Dave Airlie <airlied@redhat.com>

There have been some global drm changes between 3.14 (bochs driver added
here) and 3.15 (current drm backport level).  This patch adds the
bits for drivers/gpu/drm/bochs/ from these global changes, namely these
upstream commits:

f4510a2752b75ad5847b7935b68c233cab497f97
44d847b7439bdea0b6c5640446427daa3ebcc7fa
6796cb16c088905bf3af40548fda68c09e6f6ee5

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/bochs/bochs_kms.c b/drivers/gpu/drm/bochs/bochs_kms.c
index 62ec7d4..dcf2e55 100644
--- a/drivers/gpu/drm/bochs/bochs_kms.c
+++ b/drivers/gpu/drm/bochs/bochs_kms.c
@@ -62,10 +62,10 @@ static int bochs_crtc_mode_set_base(struct drm_crtc *crtc, int x, int y,
   }
  }
 
- if (WARN_ON(crtc->fb == NULL))
+ if (WARN_ON(crtc->primary->fb == NULL))
   return -EINVAL;
 
- bochs_fb = to_bochs_framebuffer(crtc->fb);
+ bochs_fb = to_bochs_framebuffer(crtc->primary->fb);
  bo = gem_to_bochs_bo(bochs_fb->obj);
  ret = ttm_bo_reserve(&bo->bo, true, false, false, 0);
  if (ret)
diff --git a/drivers/gpu/drm/bochs/bochs_mm.c b/drivers/gpu/drm/bochs/bochs_mm.c
index 14f9da9..1728a1b 100644
--- a/drivers/gpu/drm/bochs/bochs_mm.c
+++ b/drivers/gpu/drm/bochs/bochs_mm.c
@@ -225,7 +225,9 @@ int bochs_mm_init(struct bochs_device *bochs)
 
  ret = ttm_bo_device_init(&bochs->ttm.bdev,
      bochs->ttm.bo_global_ref.ref.object,
-     &bochs_bo_driver, DRM_FILE_PAGE_OFFSET,
+     &bochs_bo_driver,
+     bochs->dev->anon_inode->i_mapping,
+     DRM_FILE_PAGE_OFFSET,
      true);
  if (ret) {
   DRM_ERROR("Error initialising bo driver; %d\n", ret);
@@ -359,7 +361,7 @@ static int bochs_bo_create(struct drm_device *dev, int size, int align,
  }
 
  bochsbo->bo.bdev = &bochs->ttm.bdev;
- bochsbo->bo.bdev->dev_mapping = dev->dev_mapping;
+ bochsbo->bo.bdev->dev_mapping = dev->anon_inode->i_mapping;
 
  bochs_ttm_placement(bochsbo, TTM_PL_FLAG_VRAM | TTM_PL_FLAG_SYSTEM);
 
-- 
1.7.1